File: app/src/main/java/com/saitejajanjirala/imdbclone/models/Result.java
package com.saitejajanjirala.imdbclone.models;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;
import androidx.room.TypeConverter;

import org.jetbrains.annotations.NotNull;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;


@Entity(tableName = "results")
public class Result implements Serializable {
    public boolean adult;
    public String backdrop_path;
    public List<Integer> genre_ids;
    @PrimaryKey
    @NotNull
    public String pKey=""+System.currentTimeMillis();
    public int id;
    public String original_language;
    public String original_title;
    public String overview;
    public double popularity;
    public String poster_path;
    public String release_date;
    public String title;
    public boolean video;
    public double vote_average;
    public int vote_count;
    public boolean bookmarked;
    public String type;

    @Override
    public String toString() {
        return "Result{" +
                "adult=" + adult +
                ", backdrop_path='" + backdrop_path + '\'' +
                ", genre_ids=" + genre_ids +
                ", id=" + id +
                ", original_language='" + original_language + '\'' +
                ", original_title='" + original_title + '\'' +
                ", overview='" + overview + '\'' +
                ", popularity=" + popularity +
                ", poster_path='" + poster_path + '\'' +
                ", release_date='" + release_date + '\'' +
                ", title='" + title + '\'' +
                ", video=" + video +
                ", vote_average=" + vote_average +
                ", vote_count=" + vote_count +
                '}';
    }

}

File: app/src/main/java/com/saitejajanjirala/imdbclonekotlin/models/Result.kt
package com.saitejajanjirala.imdbclonekotlin.models

import androidx.room.Entity
import androidx.room.PrimaryKey
import java.io.Serializable

@Entity(tableName = "results")
data class Result  (
    var adult: Boolean = false,
    var backdrop_path: String? = null,
    var genre_ids: List<Int>? = null,
    @PrimaryKey
    var pKey: String = "" + System.currentTimeMillis(),
    var id: Int = 0,
    var original_language: String? = null,
    var original_title: String? = null,
    var overview: String? = null,
    var popularity: Double = 0.0,
    var poster_path: String? = null,
    var release_date: String? = null,
    var title: String? = null,
    var video: Boolean = false,
    var vote_average: Double = 0.0,
    var vote_count: Int = 0,
    var bookmarked: Boolean = false,
    var type: String? = null

):Serializable{

}